pipeline/join.go

```diff
@@ -90,11 +91,23 @@ func newJoinNode(e EdgeType, parents []Node) *JoinNode {
 }
 
 // Prefix names for all fields from the respective nodes.
+//
 // Each field from the parent nodes will be prefixed with the provided name and a '.'.
 // See the example above.
 //
 // The names cannot have a dot '.' character.
 //
+// If a prefix is not specified or is empty, then field names from parent nodes
+// will be copied without attaching a prefix.
+//
+// If a prefix contains a trailing '#' character, then the prefix
+// upto, but not including the trailing '#' character is prepended to the field name
+// from the parent node without adding the '.' character after the prefix.
+//
+// It is the callers responsibility to ensure that when these rules are applied
+// the collection of output field names does not contain any duplicate names. Failure
+// to ensure this will result in a runtime error.
```

Comment:
Should the entire task be stopped because of a name conflict? I know that in the `EvalNode` for example if invalid data arrives an error is logged and the task continues to process. I see two extremes

1. A single point can stop an entire task
2. A single point can be dropped form the data set.

Either extreme could be detrimental based on the use case but I think #1 is the worse of the two since is means possibly lots of data is dropped, while the task is off.